### PR TITLE
Sgmoore/update neon ext core22

### PR DESCRIPTION
**Pull request by ScarlettGatelyMoore**
- [x ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----
Needs global set for tests to pass. Store request is in.

### CODE REVIEW

**Comment by lengau** [comment]
Please fix the [linting issues](https://github.com/snapcore/snapcraft/actions/runs/4007884166/jobs/6881331224). If you've got tox 4 on your machine you can check it locally with `tox run -m lint`

File: tests/spread/extensions/kde-neon/task.yaml [hunk 1]
```diff
@@ -4,12 +4,12 @@ summary: Build and run a basic kde snap using extensions
...
 # can run on.
 systems:
-  - ubuntu-18.04
-  - ubuntu-18.04-64
-  - ubuntu-18.04-amd64
   - ubuntu-20.04
   - ubuntu-20.04-64
   - ubuntu-20.04-amd64
+  - ubuntu-22.04
+  - ubunut-22.04-64
```
Comment (lengau) [inline tests/spread/extensions/kde-neon/task.yaml:11]:
```suggestion
  - ubuntu-22.04-64
```

File: tests/unit/extensions/test_registry.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def test_get_extension_names():
     assert extensions.get_extension_names() == [
         "gnome",
         "ros2-humble",
+        "kde-neon"
```
Comment (lengau) [inline tests/unit/extensions/test_registry.py:29]:
```suggestion
        "kde-neon",
```

File: tests/spread/extensions/kde-neon/task.yaml (hunk 1 above)
Comment (ScarlettGatelyMoore) [reply tests/spread/extensions/kde-neon/task.yaml]: Done.

**Comment by ScarlettGatelyMoore** [comment]
I am pretty certain the rest of the tests require the global set on the content pack.

**Comment by sergiusens** [comment]
I have done the last bit of test updates, will keep an eye on spread integration tests. I would encourage the rename of the snaps to
- `kf5-qt%QT_VERSION%-sdk`
- `kf5-qt%QT_VERSION%`

If the kf5 version is not as important; this would save the need to rebuild all snaps all the time, save us from requesting default connections and take advantage of delta downloads for the content snap

**Comment by ScarlettGatelyMoore** [comment]
> I have done the last bit of test updates, will keep an eye on spread integration tests. I would encourage the rename of the snaps to
> 
>     * `kf5-qt%QT_VERSION%-sdk`
> 
>     * `kf5-qt%QT_VERSION%`
> 
> 
> If the kf5 version is not as important; this would save the need to rebuild all snaps all the time, save us from requesting default connections and take advantage of delta downloads for the content snap

Hi Sergio, as Harald mentioned in the channel, users would end up with weird bugs if the content snap kf5 versions don't exactly match the versions the applications are built with. So we have to keep this format. However, I can get away with shortening kde-framworks... to kde-kf5 . It won't help with the store wait time or provisional updates sadly.

**Comment by sergiusens** [comment]
@ScarlettGatelyMoore let's make sure the next update has the proper name then. Also, is the content snap clear for global connection?

**Review by sergiusens — APPROVED** [review]

**Pull request merged**